File: java/정렬/H_Index.java
package 정렬;

import java.util.Arrays;

public class H_Index {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);
        System.out.println(Arrays.toString(citations));
        int cnt = 0;
        int n = citations.length;
        for(int i = 0; i < n; i++){
            System.out.println("i = " + i + " cnt = " + cnt);
            while(true){
                if(cnt >= n - i){
                    return cnt;
                }
                if(cnt == citations[i]) break;

                cnt++;
            }
        }
        return answer;
    }
}

File: python/프로그래머스/정렬/H-Index.py
# [5,6,7] -> 5를 바라볼때 cnt가 계속 늘어나야 한다. -> while문?
def solution(citations):
    ret = 0
    n = len(citations)
    citations.sort()
    print(citations)
    
    for i in range(n):
        rest = n - i
        # print("rest = ", rest,"ret = ", ret,"citation = ", citations[i])
        while ret < citations[i] and ret < rest:
            ret += 1
            # print("ret ", ret)
    
    return ret